test/calendar.ts

```diff
@@ -0,0 +1,97 @@
+import {assert} from 'mocha-webdriver';
+import {getGrist} from "./getGrist";
+
+//not to pretty way to get events from currently used calendar control. but it's working. 
+function getAbstractFromCalendarObject(eventId: number) {
+    return `
+            const calendarObject =  this.calendarHandler.calendar.getEvent(${eventId},"cal1"); 
+            if(calendarObject)
+            {
+                return{
+                    title: calendarObject?.title,
+                    startDate: calendarObject?.start.toString(),
+                    endDate: calendarObject?.end.toString(),
+                    isAllDay: calendarObject?.isAllday??false
+                }
+            }else{
+                return calendarObject
+            }`
+}
+
+describe('calendar', function () {
+    this.timeout(20000);
+    const grist = getGrist();
+
+    before(async function () {
+        const docId = await grist.upload('test/fixtures/docs/Calendar-UnitTest.grist');
+        await grist.openDoc(docId);
+        await grist.toggleSidePanel('right', 'open');
+        await grist.addNewSection(/Custom/, /Table1/);
+        await grist.clickWidgetPane();
+        await grist.selectCustomWidget(/Calendar/);
+        await grist.setCustomWidgetAccess('full');
+        await grist.setCustomWidgetMapping('startDate', /From/);
+        await grist.setCustomWidgetMapping('endDate', /To/);
+        await grist.setCustomWidgetMapping('title', /Label/);
+        await grist.setCustomWidgetMapping('isAllDay', /IsFullDay/);
+    });
+
+    it('should create new event when new row is added', async function () {
+        await grist.sendActions([['AddRecord', 'Table1', -1, {
+            From: new Date('2023-08-03 13:00'),
+            To: new Date('2023-08-03 14:00'),
+            Label: "New Event",
+            IsFullDay: false
+        }]]);
+        const mappedObject = await grist.getCustomWidgetObject(getAbstractFromCalendarObject(1));
```

Comment:
Is the update guaranteed to have happened by the time you call `grist.getCustomWidgetObject`?